debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/RecordMakers.java

```diff
@@ -173,7 +173,7 @@ public boolean assign(long tableNumber, TableId id) {
         TableSchema tableSchema = schema.schemaFor(id);
         if (tableSchema == null) return false;
 
-        String topicName = topicSelector.getTopic(id);
+        String topicName = tableSchema.getParentSchemaName();
```

Comment:
The envelope is really the change event, which has the `before` and `after` fields that each contain the real representation of the affected row before and after the change, respectively. The envelope _wraps_ the actual data and has additional information (e.g., the `source` and event timestamp fields). 

Thus the envelope's _schema_ is simply the schema that defines the `Struct` with these fields. Note that the `before` and `after` fields have schemas that are derived from the DDL, and thus may change over time. Therefore, the envelope schema has to evolve with it, simply because it's referencing new schemas for these fields.